### DISCUSSION TITLE
Render resulting 3D Model from specific camera poses

### DISCUSSION THREAD

**Discussion by catalinad** [alicevision/Meshroom#2306] [Q&A] [answered]
Hi! Thank you for this awesome 3D reconstruction software! I am new to Meshroom and I have a couple of questions. I used Meshroom to generate a 3D model from a set of photos. I am in need of rendering the generated model from specific camera poses (the same camera poses as the input images used for generating the 3D model). Could you provide me please some guidance on how I can achieve this? Can this be done directly in Meshroom?

Thank you in advance for all your help!

**Comment by natowi** [comment]
Here is an example from the Tanksandtemples dataset in Meshroom 2023.3:

![grafik](https://github.com/alicevision/Meshroom/assets/8138537/b6bd452c-4bb1-41f3-864e-93bebdff2913)


You can overlay source image and reconstruction. Click through the Image gallery to align each image.

You could also use Blender with this addon: https://github.com/SBCV/Blender-Addon-Photogrammetry-Importer

**Reply by catalinad** [reply]
Hi! Thank you very much for your reply, it is very useful. Could you tell me please how I can save, if possible, the reconstruction images in the MeshroomCache folder? Thank you again for your support, it is highly appreciated!

**Reply by natowi** [reply]
Sorry, I don´t understand what you mean with "reconstruction images". Those are your input images as seen in the Image Viewer.
What are you trying to achieve? If you want to render the 3D model from  specific view into a file, use the Blender importer for Example and Render your view in Blender

**Reply by catalinad** [reply]
Hi, sorry for the delayed response and thank you for your advice. 
I need to render the 3D Model from specific views into files (my plan is to compute the PSNR between the source images and the rendered images). I used the Blender Addon Photogrammetry Importer and I was able to import cameras poses and textured mesh. I then rendered and saved the images using a Python script in Blender (I iterated through the reconstruction cameras and rendered the images). The problem is that when importing the reconstructed cameras in Blender, some of the camera parameters are off, e.g the FOV and Shift X and Y and because of that the rendered images are not the expected ones. With some trial and error I was able to approximate the values for these parameters, but I am concerned that by doing this will impact the accuracy of the PSNR value. 
Is there any way to save images directly from the 3D Viewer in Meshroom?
Thank you for your help!

**Reply by natowi** [reply, marked as answer]
"some of the camera parameters are off" -> I am not so familiar with Blender myself, but you should be able to create a camera with real parameters from the parameters in the metadata of your original images. Do you use the original images or the undistorted images?
Maybe check out https://github.com/tibicen/meshroom2blender as alternative? (not sure if it works with the latest release)

"Is there any way to save images directly from the 3D Viewer in Meshroom?" -> No. The 3D viewer is based on https://github.com/alicevision/QtAliceVision if you want to check out the code. Saving the view as image could be  nice contribution.

**Reply by catalinad** [reply]
Thank you @natowi setting the parameters from the metadata of the original images resolved the issue.
